### ISSUE TITLE
jhipster.github.io website design update

### ISSUE THREAD

**Issue by ntorionbearstudio** [jhipster/generator-jhipster#13595]
##### **Overview of the feature request**

This is concerning https://github.com/jhipster/jhipster.github.io

With @ascelineboullen we propose our contribution to update and refresh jHipster's website design.

In first @ascelineboullen can create wireframes / sketch on Figma to see future design and use it as a support of discussion.

We also propose to use new technologies for website development :

- [NextJs](https://nextjs.org/) for using React (to have components approach)
- [Chakra UI](https://chakra-ui.com/) as UI library for components and layout (give composable approach and it's very simple to use)

**Comment by pascalgrimaud** [comment]
A big +1 from me

I think the difficulty part would be:
- keep MD files to update content
- keep a way to test content before submitting pull request

**Comment by MarlonLuan** [comment]
If the technology has not yet been defined, I suggest you take a look at the [Hugo](https://github.com/gohugoio/hugo) it is written in go, it's simple, and according to its description it is "The world’s fastest framework for building websites."

**Comment by pascalgrimaud** [comment]
Cc @jdubois @deepu105 and @jhipster/developers

**Comment by atomfrede** [comment]
I use hugo for my personal blog and really like it as it also supports asciidoc(tor) as alternative to markdown. Before do anything we should maybe think about how to structure should be eventually. Maybe we also can make the separation between high level (marketing) overview and user documentation more clear.

**Comment by deepu105** [comment]
Yes a site revamp is heavily needed and appreciated. Few things to keep in mind (some pointers from my experience doing the last revamp)
1. We are hosted on GitHub pages that means the site needs to be Jekyll based else things will become much more difficult like having CIs that can build and publish that static content on each commit. I don't think it's a big deal but its additional work to be considered and makes deployment a bit more complex than today
2. The markdown content needs to remain the same, which means the framework (gatsby, Hugo or whatever should be able to work with existing markdown files)
3. Our markdown files use front matter for metadata and stuff like url redirect etc, this behaviours need to be retained, last I checked Gatsby and Hugo doesn't support front matter, but things might have changed
3. There are parts of the site that uses AngularJS like home page, marketplace etc, these need to be migrated, it shouldn't be a big deal IMO
4. The URL slugs should not change, this is very very important as it affects our SEO (file structure can change as long as URL slug remains the same, with Jekyll its possible with front matter not sure about others)


IMO the best way to approach this would be

1. Redesign the site CSS first
2. Move dynamic pages from Angular to React or Vue
3. Move to new SSG (Gatsby or Hugo or whatever) personally I don't care much for this step as Jekyll is still good for our usecase

**Comment by PierreBesson** [comment]
Thanks a lot for offering contributions to the design !

Having managed some aspects of the website over the years (docs archive, chinese i18n, microservice docs...). Here are my suggestion for the website:
1. Support Markdown for documentation pages. The Markdown format is the most simple to maximize ease of contributions so we need to keep it. However this is only for the documentation part, the landing page and some other specific pages could use other formats.
2. Use a static site generator (no back-end) and host in GitHub pages. On the contrary to Deepu, I would be OK to move away from Jekyll as they are a lot of other tools which are easy to setup with GitHub Action. However we really don't want to have a back-end as we have a lot of traffic from all over the world (really impressive) and don't want to pay for servers or have the website be slow for users located in far away regions.
3. Support an archive of old versions of the website (like I did for https://www.jhipster.tech/documentation-archive/)
4. Support multiple languages. As of right now we have english and chinese language website (https://jhipster.tech/cn) but it's not really an optimal setup as the Chinese website is a fork of the English version.
5. Make the doc pages look more standard. Typical tech doc pages looks like https://docs.readthedocs.io/ or https://docs.gitbook.com/. This is what we should have.

One great "framework" that fit all requirements is https://docusaurus.io/ . It is developed by the React team for their docs so I guess it could be a good base for the new website on top of which could be developed great looking design.

**Comment by ntorionbearstudio** [comment]
Hello, thanks all for your responses

1. Yes we will keep markdown files for content. With Next JS there is no problem for that : https://nextjs.org/blog/markdown
2. Yes it will be only a front-end project, no back-end is needed I think for the website
3. We will keep current URLs / slugs for all pages for SEO and archived websites
4. For languages, we will setup an internationalization system with NextJs

For now in first step, I think that we should start by designing the new website with some wireframes and then we'll go back to stack issues

**Comment by pascalgrimaud** [comment]
> For now in first step, I think that we should start by designing the new website with some wireframes and then we'll go back to stack issues

Yes plz, small steps to have quick feedbacks first :)

**Comment by ntorionbearstudio** [comment]
Hello everyone, with @ascelineboullen we started to think about new pages and links disposition.
In first, we think that some generator informations or some tools like Jhipster online generator or JDL Studio are not highlighted enough.

We'll have more time to work on it this month. We'll go back soon with some propositions to discuss with you :)

**Comment by jdubois** [comment]
Totally agree on that!
Thanks a lot for your work on this, we have a ton of visitors (110,000 unique visitors last month) and it’s going to help a lot of people.

**Comment by deepu105** [comment]
Thats a great point. i added JDL studio to main menu for now. But agree we
need a proper redesign

**Comment by atomfrede** [comment]
Keep it open.

**Comment by mraible** [comment]
I have a contact that's developing a website for us at @oktadev. I could ask them for a quote to redesign our site too. While we're add it, we should fix our vulnerabilities. I tested [Snyk's new website scanner](https://twitter.com/snyksec/status/1390350838355435520) and we get an [F](https://snyk.io/test/website-scanner/?test=210506_AiDcMZ_23d760db26ab5d0851524240262fe898&utm_medium=referral&utm_source=webpagetest&utm_campaign=website-scanner).

**Comment by jdubois** [comment]
We could at least have a new CSS, and also a custom one for the generated apps (not sure if that can work across all front-end frameworks). But having a common visual identity would be awesome.
Concerning Snyk, as the website is a static website, I wonder how we can have XSS attacks - how are those supposed to work?
Anyway, we have money and this is the kind of thing we need to pay for, so yes @mraible if you can get a quote that's very interesting.

**Comment by ntorionbearstudio** [comment]
Hello @mraible @jdubois, @ascelineboullen is preparing some sketchs for the jHipster website, we'll be able to show you this next week I think. This can be a base for a reflexion about the website reorganization and it will give some ideas about the update of the design

**Comment by CGarces** [comment]
> Concerning Snyk, as the website is a static website, I wonder how we can have XSS attacks - how are those supposed to work?
> Anyway, we have money and this is the kind of thing we need to pay for, so yes @mraible if you can get a quote that's very interesting.

Google.com gets a "D" on Snyk... just for reference.

**Comment by ntorionbearstudio** [comment]
@ascelineboullen did a first version of wireframes for the website design update : https://rebrand.ly/jhipster !
There are still some parts in progress, but you can start to send us feedbacks on the reorganization and on the design choices :)

**Comment by deepu105** [comment]
Home page looks great. We need to move the sponsor section up though

**Comment by mraible** [comment]
@ntorionbearstudio I ❤️ it! It looks fantastic. Do you plan to offer light and dark options?

**Comment by jdubois** [comment]
Yes @ntorionbearstudio this looks awesome! I especially like the docs, this really needs to be improved.

**Comment by DanielFran** [comment]
Great @ntorionbearstudio seems awesome

**Comment by pascalgrimaud** [comment]
Still in progress

**Comment by DanielFran** [comment]
Still in progress

**Comment by ascelineboullen** [comment]
Keep it open.
Summer is ending, we will be able to resynchronize with @ntorionbearstudio, and focus, to take back this project and do propositions ! 😀💪🏻

**Comment by ascelineboullen** [comment]
Thank you @mraible , we will take a look with @ntorionbearstudio  ! :)

**Comment by ascelineboullen** [comment]
Hello ! For your information, I've worked on, this week end. I will update the Figma link this week, when it will be good and ready to take your feedbacks ! :)

**Comment by pascalgrimaud** [comment]
Thanks @ascelineboullen !
Can't wait :-)

**Comment by jdubois** [comment]
That's awesome @ascelineboullen ! I'm also eager to see this!

**Comment by ascelineboullen** [comment]
Hello ! 
I just updated the Figma link https://rebrand.ly/jhipster. 
I have to do some more margin / icons and photos adjustments and it will be good, but you can already give your feedback ;). Thanks !

**Comment by pascalgrimaud** [comment]
Nice work @ascelineboullen : love the new design
I have only 1 feedback at the moment: the site needs to be responsive, but I'm pretty sure it will be

**Comment by MathieuAA** [comment]
That's awesome! Nice job @ascelineboullen!!

**Comment by deepu105** [comment]
Excellent work! :clap: 

Few small comments:

- I would put teams in a separate page as we have today and keep modules/blueprints at the end of the page like today
- I think we should retain the quick start section

**Comment by avdev4j** [comment]
I really like the new logo with the JHipster family and the whole branding.
The colors are great,

Maybe I would put sponsors before companies that use JHipster.

Anyway, you find the best solution to release more frequently by cloning
Pascal. Good job!

**Comment by jdubois** [comment]
Yes it looks awesome! I mostly have comments about the content:

- Like @avdev4j I'd rather have the sponsors before the users (the sponsors support us truly with $$$, the users are usually free riders)
- Their is some wording issues: it's "JHipster", not "Jhipster"
- I'm not sure about the news section: it's awesome to show our events, but this needs to be updated frequently and we're usually bad at this
- We should have start.jhipster.tech focused early, as well as some "getting started" things like the videos from @mraible - that's the first thing people want, what does it looks like it and how to start using it

**Comment by wmarques** [comment]
Really like the new design, awesome job 👏

**Comment by colameo** [comment]
congrats! really cool :-)

**Comment by mraible** [comment]
This looks fantastic!

A couple typos I noticed:

- JHipster online should be JHipster Online
- Github should be GitHub
- Grettings Java JHipster should be Greetings Java Hipster

**Comment by ascelineboullen** [comment]
Thanks for your feedbacks ! 😀
I will rework on it this week end again to make somes fixs, and apply your differents feedbacks, and I will design a responsive view to show you how it could be on a mobile, yes @pascalgrimaud , no worries !

**Comment by deepu105** [comment]
@ascelineboullen can I make a request for the family image. Can you please include more diverse characters from the collection there as there are too many white characters in this and we want a better mix

**Comment by ascelineboullen** [comment]
@deepu105 , yes I were talking about it with @avdev4j. 
Actually, I'm still working on the responsive version, but I will do that asap, with the differents fixs needed! ;). 
I continue!

**Comment by pascalgrimaud** [comment]
work in progress

**Comment by ascelineboullen** [comment]
Thank you @pascalgrimaud for the message to stop the bot, I missed it! I had COVID and it was a little bit difficult this last weeks... 🙏 

For the news: I I have not finished yet but I fixed some issues that you noticed me (illustration for example) and I'm working also on the responsive view. I will do an update soon (Probably this week end)! 😃

**Comment by pascalgrimaud** [comment]
oh no worry @ascelineboullen 
Take care of yourself, it's more important than Open Source project.

There is no deadline in Open Source project, so take the time you need, really.

**Comment by ascelineboullen** [comment]
Thank @pascalgrimaud, I know, but now I'm ok, so I will be able to continue on it! 😄

**Comment by ascelineboullen** [comment]
For information, I updated the mockup: I fixed all the wording errors that you reported, thanks 😃 .

Next: I'm still thinking about your feedback on the structure of the pages (split the team part on another page, how to design the getting started, the news, etc...). In parallel I continue on the responsive version. 

I will do another update soon !

**Comment by jdubois** [comment]
Thank you @ascelineboullen for your hard work!

**Comment by ascelineboullen** [comment]
Hello everyone!  

A little update: 
- On desktop version (https://rebrand.ly/jhipster), I fixed some elements on the homepage (I deleted "news section", I split "team section" on another page, fixed somes mistakes on texts, etc..)
- It's not finished yet, but I init the link to see the homepage on the responsive view. You can see it at: https://rebrand.ly/jhipster-mobile

So don't hesitate to give me some feedback! 

I continue on the mockup 😃

**Comment by jdubois** [comment]
Thank you for your hard work @ascelineboullen ! As I'm currently learning more about accessibility, can you have a look at https://accessibilityinsights.io to be sure that the new website would work for everyone?

**Comment by ascelineboullen** [comment]
Yes I design keeping accessibility constraints in mind. But I will take a look at the link you shared with me to reconfirm that's really ok. Thank @jdubois 😄

**Comment by mraible** [comment]
Can you please change the test from "Grettings Java Hipster" to "Greetings Java Hipster" at the top?

**Comment by ascelineboullen** [comment]
Yes @mraible, I just fixed to "Greetings Java Hipster". Sorry I missed it, thank! :)

**Comment by ascelineboullen** [comment]
Hello everyone! 
I will work on this issue this week, and I will do an update this week end 😄

**Comment by ascelineboullen** [comment]
**New little update:**
About desktop proposition [https://rebrand.ly/jhipster](https://rebrand.ly/jhipster)
- I've modified/re-organized the structure and hierarchy for main page,
- split team section on another page (work in progress),
- add the quick start section with the video, 
- and also add blueprints (at the end of the webpage. Bloc is in progress, I will modified the design if the structure is ok).

To answer you @jdubois : thanks I've check the link you shared about accessibility, it's seems more for the development step, but it was interesting, thank you! Yes, to make the revamp of this webpage, I considered rules of design and accessibility (and Figma offers plugins to test accessibility, such as plugins related to color contrasts for example). Also, about it, today, I open another issue in parallel to this issue, about the brand identity and guidelines (https://github.com/jhipster/jhipster-artwork/issues/21). To complete the graphical charter, and define typefaces for the content, colors rules, etc. It will help us to have established rules (accessibility contrast), and then to have a consistent and coherent whole in the projects. 

Also, for your information, I will have more time to allocate to contributions, so I continue on the mockup and additions to the graphical charter, but you can already give me some feedbacks again! 😉. About the structure, what did you think? 
 
Thanks!

**Comment by mraible** [comment]
It looks awesome! Nicely done @ascelineboullen!!

**Comment by deepu105** [comment]
Looks great. Well done 👏

**Comment by jdubois** [comment]
Yes that's great work!!! We also need better UX/UI for JHipster Lite, @pascalgrimaud

**Comment by ascelineboullen** [comment]
Thanks for your positive feedback! 🤩. I will finished module section & team page, and then I will modify the responsive view (on mobile) for the homepage. 

Yes, we've discussed with Pascal about JHipster Lite and the better UX/UI needed. That's why I've opened an issue about graphical charter, to have a common graphic base defined before going further in the differents projects. And then, yes I could help in JHipster Lite's project too ;)

**Comment by ascelineboullen** [comment]
Little update: 
Tonight, I'm working on the module section & team page, I will be able to update the link tomorrow and bring fews news 🙂.

_I worked the last week on https://github.com/jhipster/jhipster-artwork/issues/21 too. For this issue, I will make an update this week end and show you the draft of the proposal that I want to make for the update of the graphical charter!_

**Comment by ascelineboullen** [comment]
**UPDATE:**

**Desktop version (https://rebrand.ly/jhipster):** 
- I've added section modules at the bottom of the homepage
- I've updated team page: now, all sections are there

**Mobile version (https://rebrand.ly/jhipster-mobile)**
- Mobile homepage is now up to date with homepage desktop version 

Then, I will work to fix some details: add various datas (because members are still called "Pascal" for exemple 😄. Same problem for tech logo), to be more realistic and get closer to the version that will be integrated.

Feel free to give me more feedback! 🙂.
Thanks!

**Comment by Jessy-BAER** [comment]
If the wireframe are good enough, I propose to work on the implementation with asceline. 

Maybe it's better to create a specific issue for it ?

**Comment by ascelineboullen** [comment]
I agree with Jessy. The base of the redesign seems to be ok, here is the export of the first version of the mockup. ;)

![Landing page - mobile](https://user-images.githubusercontent.com/9590731/218448026-fb575cdf-313f-4167-9f24-a5f92aaa9869.jpg)
![Landing Page - Team](https://user-images.githubusercontent.com/9590731/218448055-37bb53d8-5f82-44f7-a168-cb31197a80ad.jpg)
![Landing Page - Home](https://user-images.githubusercontent.com/9590731/218448066-7102c1cb-64d7-4203-b4b0-545d7cc3edec.jpg)

We will create more specific tickets as Jessy indicated, to go on the implementation.

I've init a document to think about the site tree, we're going to work on it with Jessy to confirm the structure of the new project.
https://whimsical.com/jhipster-arborescence-RxvQrWktaDY9P2DTzSSR9P

**Comment by deepu105** [comment]
Looks really good. I would suggest to implement the Visual design in
CSS/Sass first without migrating from Jekyll so that we have something to
start with and once visual design is in place we could consider further
migrations depending on who is interested to do it :P

Thanks & Regards,
Deepu

**Comment by Jessy-BAER** [comment]
@deepu105 I am on it [21164](https://github.com/jhipster/generator-jhipster/issues/21164), but i replace bootstrap by tailwindcss you can see [my work](https://github.com/Jessy-BAER/jhipster.github.io/pull/1) ^^

**Comment by irwansyahwii** [comment]
I just got laid off, I can work on this full time. Can I take it?

**Comment by pascalgrimaud** [comment]
The design was done and the bounty already claimed in another ticket

We'll open new ticket to improve the new version
